src/cluster/kind/mod.rs

```diff
@@ -0,0 +1,159 @@
+use std::{
+    io::Write,
+    process::{Command, Stdio},
+};
+
+use snafu::{ensure, ResultExt, Snafu};
+use tracing::{debug, info, instrument};
+
+use crate::{
+    cluster::{check_if_docker_is_running, kind::config::KindClusterConfig, DockerError},
+    constants::{DEFAULT_LOCAL_CLUSTER_NAME, DEFAULT_STACKABLE_NAMESPACE},
+    utils::check::binaries_present,
+};
+
+mod config;
+
+#[derive(Debug, Snafu)]
+pub enum ClusterError {
+    #[snafu(display("io error: {source}"))]
+    IoError { source: std::io::Error },
+
+    #[snafu(display("stdin error"))]
+    StdinError,
+
+    #[snafu(display("command error: {error}"))]
+    CmdError { error: String },
+
+    #[snafu(display("missing dependencies"))]
+    MissingDepsError,
+
+    #[snafu(display("Docker error: {source}"))]
+    DockerError { source: DockerError },
+
+    #[snafu(display("yaml error: {source}"))]
+    YamlError { source: serde_yaml::Error },
+}
+
+#[derive(Debug)]
+pub struct KindCluster {
+    cp_node_count: usize,
+    namespace: String,
+    node_count: usize,
+    name: String,
+}
+
+impl KindCluster {
+    /// Create a new kind cluster. This will NOT yet create the cluster on the system, but instead will return a data
+    /// structure representing the cluster. To actually create the cluster, the `create` method must be called.
+    pub fn new(
+        node_count: usize,
+        cp_node_count: usize,
+        name: Option<String>,
+        namespace: Option<String>,
+    ) -> Self {
+        Self {
+            namespace: namespace.unwrap_or(DEFAULT_STACKABLE_NAMESPACE.into()),
+            name: name.unwrap_or(DEFAULT_LOCAL_CLUSTER_NAME.into()),
+            cp_node_count,
+            node_count,
+        }
+    }
+
+    /// Create a new local cluster by calling the kind binary.
+    #[instrument]
+    pub fn create(&self) -> Result<(), ClusterError> {
```

Comment:
Arguably this should be async (using `tokio::process` rather than `std::process`).